### DISCUSSION TITLE
How to add retrofit to investments

### DISCUSSION THREAD

**Discussion by jkiviluo** [spine-tools/SpineOpt.jl#875] [Ideas]
Different options:
- Change source (gas to H2)
- Extend lifetime

Possibly can already be done with user constraints on investments (but not very easy)

**Comment by nnhjy** [comment]
For units, connections, and storage (maybe?). User constraints can basically work on units (thanks to the `unit_on` variable).

More options:
- Decommission (existing `units_mothballed` would be useful, mentioned in [this issue](https://github.com/spine-tools/SpineOpt.jl/issues/561#issue-1614044473))
